### PR TITLE
Add SortByResponse and LabelEncoding categorical_encoding schemes.

### PR DESCRIPTION
**Pull request by arnocandel**
For DRF/GBM, we now sort all categorical columns with more than nbins_cats factors, to allow the <= split to be useful and to reduce the effective number of factors in the child nodes. Otherwise, it would quite likely never split on this feature, and the bitset magic would never occur. Now, we get higher (over)fitting ability.

### CODE REVIEW

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
The consensus between @mlandry22 @IzuiT and me was that we shouldn't make this behavior the new default - the potential for overfitting is too large. Better to let the user specify a larger nbins_cats if they want to increase the fitting capability for high-cardinality features.

**Comment by arnocandel** [comment]
Now we have two new categorical_encoding schemes: Integer and SortByResponse.

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/hex/Model.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public double defaultThreshold() {
       AUTO, Random, Modulo, Stratified
     }
     public enum CategoricalEncodingScheme {
-      AUTO, OneHotInternal, OneHotExplicit, Enum, Binary, Eigen
+      AUTO, OneHotInternal, OneHotExplicit, Enum, Binary, Eigen, Integer, SortByResponse
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/Model.java]: Is it OK to call it `Integer`? It sounds like `java.lang.Integer`

File: h2o-core/src/main/java/hex/ModelBuilder.java [hunk 2]
```diff
@@ -981,6 +982,36 @@ else if (_parms._weights_column != null && _weights != null && !_weights.isBinar
...
+          Vec vNew = new VecUtils.ReorderTask(invIdx).doAll(1, Vec.T_NUM, new Frame(v)).outputFrame().anyVec();
+          String[] newDomain = new String[len];
+          for (int i = 0; i < len; ++i) newDomain[i] = v.domain()[idx[i]];
+          vNew.setDomain(newDomain);
+//          for (int i=0;i<len;++i)
+//            Log.info(vNew.domain()[i] + " -> " + meanWeightedResponse[idx[i]]);
+          vecs[j] = vNew;
+        }
+      }
+      _train.restructure(_train.names(), vecs);
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/ModelBuilder.java]: Arno, do you need to call it even nothing was changed?

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Pull request merged**